src/sentry/grouping/ingest/config.py

```diff
@@ -108,7 +99,10 @@ def project_uses_optimized_grouping(project: Project) -> bool:
     secondary_grouping_config = project.get_option("sentry:secondary_grouping_config")
     has_mobile_config = "mobile:2021-02-12" in [primary_grouping_config, secondary_grouping_config]
 
-    return not has_mobile_config and features.has(
+    if has_mobile_config:
+        return False
```

Comment:
Indeed. As you can see, the restriction has always been there (because the new path has none of the hierarchical-hash-handling code), but it seemed clearer to split it out now that the criteria are more complicated.